src/Datadog.Trace/Util/PropertyFetcher.cs

```diff
@@ -10,7 +9,7 @@ internal class PropertyFetcher
     {
         private readonly string _propertyName;
         private Type _expectedType;
-        private PropertyFetch _fetchForExpectedType;
+        private object _fetchForExpectedType;
```

Comment:
It bothers me a bit. I'll probably come back to it at some point to try and remove the cast